File: src/test/java/ukalus/container/creators/ClassMethodCreatorTest.java
package ukalus.container.creators;

import ukalus.container.InstanceCreator;
import ukalus.container.ObjectRegistry;
import ukalus.container.Resolver;
import ukalus.container.resolvers.ValueResolver;

import java.lang.reflect.InvocationTargetException;
import java.net.InetAddress;
import java.net.UnknownHostException;
import java.nio.ByteBuffer;

import junit.framework.TestCase;

public class ClassMethodCreatorTest extends TestCase {

  public void test0ArgMethod() throws IllegalArgumentException,
      InstantiationException, IllegalAccessException,
      InvocationTargetException, UnknownHostException {
    InetAddress expected = InetAddress.getLocalHost();
    InstanceCreator creator = new ClassMethodCreator(InetAddress.class,
      "getLocalHost", new Resolver[0]);
    Object actual = creator.newInstance();
    assertEquals(expected, actual);
  }

  public void test1ArgMethod() throws IllegalArgumentException,
      InstantiationException, IllegalAccessException,
      InvocationTargetException, UnknownHostException {
    InetAddress expected = InetAddress.getByName("www.google.com");
    InstanceCreator creator = new ClassMethodCreator(InetAddress.class,
      "getByName", new Resolver[]{new ValueResolver(new ObjectRegistry(),
        "www.google.com")});
    Object actual = creator.newInstance();
    assertEquals(expected, actual);
  }

  public void test1IntArgMethod() throws IllegalArgumentException,
      InstantiationException, IllegalAccessException,
      InvocationTargetException, UnknownHostException {
    ByteBuffer expected = ByteBuffer.allocate(1024);
    InstanceCreator creator = new ClassMethodCreator(ByteBuffer.class,
      "allocate",
      new Resolver[]{new ValueResolver(new ObjectRegistry(), "1024")});
    ByteBuffer actual = (ByteBuffer) creator.newInstance();
    assertEquals(expected.capacity(), actual.capacity());
  }

}

File: src/test/java/ukalus/container/creators/ClassMethodCreatorTest.kt
package ukalus.container.creators

import junit.framework.TestCase
import ukalus.container.ObjectRegistry
import ukalus.container.Resolver
import ukalus.container.resolvers.ValueResolver
import java.lang.reflect.InvocationTargetException
import java.net.InetAddress
import java.net.UnknownHostException
import java.nio.ByteBuffer

class ClassMethodCreatorTest : TestCase() {

    @Throws(IllegalArgumentException::class, InstantiationException::class, IllegalAccessException::class, InvocationTargetException::class, UnknownHostException::class)
    fun test0ArgMethod() {
        val expected = InetAddress.getLocalHost()
        val creator = ClassMethodCreator(InetAddress::class.java,
                "getLocalHost", arrayOf<Resolver>())
        val actual = creator.newInstance()
        TestCase.assertEquals(expected, actual)
    }

    @Throws(IllegalArgumentException::class, InstantiationException::class, IllegalAccessException::class, InvocationTargetException::class, UnknownHostException::class)
    fun test1ArgMethod() {
        val expected = InetAddress.getByName("www.google.com")
        val creator = ClassMethodCreator(InetAddress::class.java,
                "getByName", arrayOf<Resolver>(ValueResolver(ObjectRegistry(),
                "www.google.com")))
        val actual = creator.newInstance()
        TestCase.assertEquals(expected, actual)
    }

    @Throws(IllegalArgumentException::class, InstantiationException::class, IllegalAccessException::class, InvocationTargetException::class, UnknownHostException::class)
    fun test1IntArgMethod() {
        val expected = ByteBuffer.allocate(1024)
        val creator = ClassMethodCreator(ByteBuffer::class.java,
                "allocate",
                arrayOf<Resolver>(ValueResolver(ObjectRegistry(), "1024")))
        val actual = creator.newInstance() as ByteBuffer
        TestCase.assertEquals(expected.capacity(), actual.capacity())
    }

}
